extensions/wikia/FandomCreator/FandomCreator.setup.php

```diff
@@ -0,0 +1,18 @@
+<?php
+spl_autoload_register(function($class) {
+	$prefix = 'FandomCreator\\';
+	$len = strlen($prefix);
+	if (strncmp($prefix, $class, $len) !== 0) {
+		return;
+	}
+
+	$file = __DIR__.'/'.str_replace('\\', '/', substr($class, $len)).'.php';
+	if (file_exists($file)) {
+		require $file;
+	}
+});
+
+$wgHooks['NavigationApigetDataAfterExecute'][] = function(WikiaController $controller) {
```

Comment:
Where does this hook get called? Didn't find it with `ack`.

Reply:
That's Nirvana magic dispatching https://github.com/Wikia/app/blob/dev/includes/wikia/nirvana/WikiaDispatcher.class.php#L231

Reply:
Yeah I'm torn between adding individual hooks for each of the api responses we'll have to intercept so it's easier to grep for, or taking advantage of the magic that's already there. For now I chose to do it this way so that I don't have to touch any code outside of our extension.

Reply:
Thanks for the tip!

Reply:
@nmonterroso By the way I think its better to use `WikiaDispatchableObject` as the type hint in case one of the controllers get changed to `WikiaService` for security or other reasons. I ran into this gotcha a while ago in Thumbnails extension.

Reply:
Good call. `WikiaDispatchableObject` still gives access to `WikiaResponse` which is what we need. cb888a1